src/bindings/jna/libelektra4j/src/main/java/org/libelektra/Plugin.java

```diff
@@ -14,7 +22,7 @@
 	 *            a key
 	 * @return the plugin's return value for open
 	 */
-	int open(KeySet conf, Key errorKey);
+	int kdbOpen(KeySet conf, Key errorKey);
```

Comment:
Why do you rename here? The prefix seems unnecessary (both for Plugin and PluginLoader).

It would be great if you can add a test where a Java plugin is loaded and tested.

Reply:
The names are the equivalents to those in the C written implementation. I wanted it to be consistent but if you wish I can refactor it.

Reply:
> It would be great if you can add a test where a Java plugin is loaded and tested.

I do not have written those plugins and also do not have the time to test them unfortunately.

Reply:
> The names are the equivalents to those in the C written implementation. I wanted it to be consistent but if you wish I can refactor it.

It is astonishing that such important decisions are not mentioned at all. **Please** document such things, it would have saved a lot of time for both of us.

src/libs/tools/include/plugin.hpp is not really very good design, but it also avoid the "kdb" prefixes.

Instead of the modules as parameter it is nicer to have a PluginLoader, which is a very usual thing in Java.

Reply:
Furthermore, it would be nice if one gets a "Plugin" in both cases, as the NativePlugin and the current "Plugin" obviously should be subclass of a common class.

Reply:
> It is astonishing that such important decisions are not mentioned at all. Please document such things, it would have saved a lot of time for both of us.

Why is it an important decision how I name a method? I will simply revert it to the  non-prefix version.

> Instead of the modules as parameter it is nicer to have a PluginLoader, which is a very usual thing in Java.

You mean you want some sort of "Factory"? The current NativePlugin is just a proxy to abstract the pointer logic away and implement the Plugin Interface correctly.

> Furthermore, it would be nice if one gets a "Plugin" in both cases, as the NativePlugin and the current "Plugin" obviously should be subclass of a common class.

Currently the `Plugin` class is an interface and the NativePlugin along with the other plugins implement it. I do not know what you mean here.

Reply:
>  Why is it an important decision how I name a method? I will simply revert it to the non-prefix version.

It is not only about the method names: it was completely unclear me what the whole NativeElektraPlugin is about. The renaming of the public methods is also problematic: it breaks all already written plugins.

> You mean you want some sort of "Factory"?

Yes, something like https://docs.oracle.com/javase/7/docs/api/java/lang/ClassLoader.html

> The current NativePlugin is just a proxy to abstract the pointer logic away and implement the Plugin Interface correctly.

Which does not say anything about if this design is good.

>  Currently the Plugin class is an interface and the NativePlugin along with the other plugins implement it. I do not know what you mean here.

I was talking about the ClassLoader: it could be a secret that such a thing like a NativePlugin exists at all.

Reply:
I am still not sure how exactly how I should name the NativeElektraPlugin as it has to have a name. I can make a PluginLoader but still I need to have name for the class that loads the NativePlugins from elektra and not those implemented in java

Reply:
NativePlugin or Plugin in another package should be ok.